src/flowgger/encoder/passthrough_encoder.rs

```diff
@@ -28,7 +28,11 @@ impl Encoder for PassthroughEncoder {
             if self.header_time_format.is_some() {
                 let ts = match build_prepend_ts(self.header_time_format.as_ref().unwrap()) {
                     Ok(ts) => ts,
-                    Err(_) => return Err("Cannot prepend str"),
+                    Err(_) => {
+                        return Err(
+                            "Failed to format date when building prepend timestamp for header",
```

Comment:
it's better but the point is to have a message explaining the issue, if we have the same message from different functions it means the error is not clear (e.g. Failed to format datetime for passthrough header / Failed to format datetime for rfc3164 header) . 

This may seem trivial, but when we got that and have 2 different format, we’ll be happy to see the difference


Also in my initial comment i was taking this as an example and didn’t list all error messages I assumed you would look at the error handling as a whole. 

We have the same msg “Unable to parse the date” in 10 instances in 10 different functions (it was already not clear as we had 3 before your changes though, it is just worse). A single word added to the string could bring much better context as to what date/format/protocol we're talking about. Sometime even the function name would bring the necessary context.